src/Pages/Applications/DevtronApps/service.types.ts

```diff
@@ -27,3 +27,30 @@ export interface AppConfigStatusResponseItem {
 }
 
 export type AppConfigStatusResponse = ResponseType<AppConfigStatusResponseItem[]>
+
+export enum ResourceConfigState {
+    Unnamed,
+    Draft,
+    ApprovalPending,
+    Published,
+}
+
+export enum ResourceType {
+    ConfigMap = 'ConfigMap',
+    Secret = 'Secret',
+    DeploymentTemplate = 'Deployment Template',
+}
+
+export interface ResourceConfig {
+    name: string
+    configState: ResourceConfigState
+    type: ResourceType
+    overridden: boolean
+    global: boolean
+}
+
+export interface EnvConfig {
+    resourceConfig: ResourceConfig[]
+}
+
+export type EnvConfigResponse = ResponseType<EnvConfig>
```

Comment:
1. EnvConfig should be EnvConfigType
2. We in general don't create transitive response types. We can use ResponseType<EnvConfig> directly wherever applicable